From 7a622a11b2be157a7b006e30c90848411eae0ca5 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 22 Oct 2018 10:13:22 +0300
Subject: [PATCH 0721/1051] ARM64: dts: armada-8040-db-H: Add dual PCIe x4
 board setup H

Add DTS file for Armada-8040-DB setup H
This configuration includes 1 PCIe x4 on SERDES lanes 0-3,
one SPI+ 10Gb port on SERDES lane 4 and one PCIe x1 port
on SERDES lane 5 in each of CP0 and CP1.
This setup requeres usage of second generation IO board:
SLM-1521-V2

Change-Id: I5907d79b1a1cae629c59a86a2acdc0704ab7ccfe
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60428
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile          |   1 +
 .../boot/dts/marvell/armada-8040-db-H.dts     | 143 ++++++++++++++++++
 2 files changed, 144 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-H.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 3a96b5c1ca34..ba0dbefe3632 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -15,6 +15,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-E.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-G.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-H.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-single-shot.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8080-pd.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-H.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-H.dts
new file mode 100644
index 000000000000..1caac6a006b7
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-H.dts
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Device Tree file for Marvell Armada 8040 development board
+ * This file supports option B configuration
+ * AP_UART,AP_eMMC,CP0_I2C,CP0_eMMC,CP0_RGMII1,CP1_SPI(Boot),CP1_RGMII0
+ */
+
+#include "armada-8040-db.dtsi"
+
+/ {
+	model = "Marvell Armada 8040 development board H setup";
+	compatible = "marvell,armada8040-db-B", "marvell,armada8040-db",
+		"marvell,armada8040", "marvell,armada-ap806-quad",
+		"marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth2;
+		ethernet2 = &cp1_eth0;
+		ethernet3 = &cp1_eth1;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	status = "okay";
+};
+
+&cp0_sdhci0 {
+	status = "okay";
+};
+
+/* CON6 on CP0 expansion */
+&cp0_pcie0 {
+	num-lanes = <4>;
+	status = "okay";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy0 0
+		&cp0_comphy1 0
+		&cp0_comphy2 0
+		&cp0_comphy3 0>;
+};
+
+/* CON5 on CP0 expansion */
+&cp0_pcie2 {
+	status = "okay";
+	phys = <&cp0_comphy5 2>;
+};
+
+&cp0_mdio {
+	status = "okay";
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth1>;
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
+
+&cp1_spi1 {
+	status = "okay";
+};
+
+/* CON6 on CP1 expansion */
+&cp1_pcie0 {
+	status = "okay";
+	phys = <&cp1_comphy0 0>;
+};
+
+/* CON7 on CP1 expansion */
+&cp1_pcie1 {
+	status = "okay";
+	phys = <&cp1_comphy4 1>;
+};
+
+/* CON5 on CP1 expansion */
+&cp1_pcie2 {
+	status = "okay";
+	phys = <&cp1_comphy5 2>;
+};
+
+&cp1_mdio {
+	status = "okay";
+
+	cp1_phy1: ethernet-phy@2 {
+		reg = <0>;
+	};
+};
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+&cp1_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth1>;
+};
+
+&cp1_eth1 {
+	status = "okay";
+	phy = <&cp1_phy1>;
+	phy-mode = "rgmii-id";
+};
+
+&cp0_crypto {
+	status = "okay";
+};
+
+&cp1_crypto {
+	status = "okay";
+};
-- 
2.17.1

